gitfile/checkout.go

```diff
@@ -77,17 +80,24 @@ func CheckoutCreate(d *schema.ResourceData, meta interface{}) error {
 
 func CheckoutRead(d *schema.ResourceData, meta interface{}) error {
 	checkout_dir := d.Id()
+	repo := d.Get("repo").(string)
+	branch := d.Get("branch").(string)
 
 	if _, err := os.Stat(checkout_dir); err != nil {
-		d.SetId("")
-		return nil
+		log.Printf("checkout dir does not exist: %s", checkout_dir)
+		log.Printf("cloning repository [%s] to [%s]", repo, checkout_dir)
+
+		if err := os.MkdirAll(checkout_dir, 0755); err != nil {
+			return err
+		}
+		if _, err := gitCommand(checkout_dir, "clone", "-b", branch, "--", repo, "."); err != nil {
```

Comment:
Potential fix 2: check out the repository during the plan step (if it's not checked out).